src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/InteropServices/PlatformCompatibilityAnalyzer.Data.cs

```diff
@@ -22,14 +23,28 @@ public sealed partial class PlatformCompatibilityAnalyzer
         ///  - UnsupportedFirst - keeps the lowest version of [UnsupportedOSPlatform] attribute found
         ///  - UnsupportedSecond - keeps the second lowest version of [UnsupportedOSPlatform] attribute found
         /// </summary>
-        private class PlatformAttributes
+        private class Versions
         {
             public Version? SupportedFirst { get; set; }
             public Version? SupportedSecond { get; set; }
             public Version? UnsupportedFirst { get; set; }
             public Version? UnsupportedSecond { get; set; }
-            public bool HasAttribute() => SupportedFirst != null || UnsupportedFirst != null ||
+            public bool IsSet() => SupportedFirst != null || UnsupportedFirst != null ||
                         SupportedSecond != null || UnsupportedSecond != null;
         }
+
+        private class PlatformAttributes
+        {
+            public PlatformAttributes() { }
+
+            public PlatformAttributes(Callsite callsite, SmallDictionary<string, Versions> platforms)
+            {
+                Callsite = callsite;
```

Comment:
Because of attributes kept in the cache i needed to add Call-site info for subsequent calls made within that call site